tests/test_image_types.py

```diff
@@ -0,0 +1,346 @@
+                srcSet(sizes: [200], format: "webp")
+            }
+        }
+        """
+        results = self.client.execute(query, variables={"id": self.example_image.id})
+        self.assertEqual("", results["data"]["image"]["srcSet"])
+
+    @override_settings(GRAPPLE={"ALLOWED_IMAGE_FILTERS": ["width-200|format-webp"]})
+    def test_src_set_allowed_filter(self):
+        query = """
+        query ($id: ID!) {
+            image(id: $id) {
+                srcSet(sizes: [200], format: "webp")
+            }
+        }
+        """
+        results = self.client.execute(query, variables={"id": self.example_image.id})
+        self.assertIn("width-200.format-webp.webp", results["data"]["image"]["srcSet"])
+
+    def test_rendition_allowed_method(self):
+        self.assertTrue(rendition_allowed("width-100"))
+        with override_settings(GRAPPLE={"ALLOWED_IMAGE_FILTERS": ["width-200"]}):
+            self.assertFalse(rendition_allowed("width-100"))
+            self.assertTrue(rendition_allowed("width-200"))
+
+        with override_settings(GRAPPLE={"ALLOWED_IMAGE_FILTERS": []}):
+            self.assertFalse(rendition_allowed("width-100"))
+            self.assertFalse(rendition_allowed("fill-100x100"))
+
+    def test_src_set_num_queries(self):
+        sizes = [360, 720, 1024]
+        filters = [f"width-{size}" for size in sizes]
+
+        def get_renditions(image):
+            for img_filter in filters:
+                image.get_rendition(img_filter)
+
+        # Generate renditions for each filter in the filters list for all images
+        get_renditions(self.example_image)
+        for i in range(4):
+            get_renditions(wagtail_factories.ImageFactory(title=f"Image {i}"))
+
+        query = """
+        {
+            images {
+                srcSet(sizes: [360, 720, 1024])
+            }
+        }
+        """
+
+        with self.assertNumQueries(2):
+            self.client.execute(query)
+
+    def test_schema_for_svg_related_fields_and_arguments(self):
+        results = self.introspect_schema_by_type(self.image_model.__name__)
+        mapping = {
+            field["name"]: field for field in results["data"]["__type"]["fields"]
+        }
+        rendition_args = {arg["name"]: arg for arg in mapping["rendition"]["args"]}
+
+        if WAGTAIL_VERSION >= (5, 0):
+            self.assertIn("isSvg", mapping)
+            self.assertEqual(mapping["isSvg"]["type"]["kind"], "NON_NULL")
+            self.assertEqual(rendition_args["preserveSvg"]["type"]["name"], "Boolean")
+            self.assertEqual(
+                rendition_args["preserveSvg"]["description"],
+                "Prevents raster image operations (e.g. `format-webp`, `bgcolor`, etc.) being applied to SVGs. "
+                "More info: https://docs.wagtail.org/en/stable/topics/images.html#svg-images",
+            )
+        else:
+            self.assertNotIn("isSvg", mapping)
+            self.assertNotIn("preserveSvg", rendition_args)
```

Comment:
**Issue:** A test should only have one expected result.

Can this be placed within `ImageTypesTestWithSVG` with all the other Wagtail5+ SVG tests?

Then we can alter this to:

```suggestion
    @unittest.skipIf(WAGTAIL_VERSION >= (5, 0), "Not applicable to Wagtail versions 5.0+"
    def test_svg_fields_not_in_schema(self):
        """
        Ensure SVG-related fields and arguments are not included within the schema if we're testing against Wagtail < 5.0.
        """
        
        results = self.introspect_schema_by_type(self.image_model.__name__)
        mapping = {
            field["name"]: field for field in results["data"]["__type"]["fields"]
        }
        rendition_args = {arg["name"]: arg for arg in mapping["rendition"]["args"]}
        
        self.assertNotIn("isSvg", mapping)
        self.assertNotIn("preserveSvg", rendition_args)
```

https://docs.python.org/3/library/unittest.html#unittest.skipIf